File: 02.実践帳票サンプル集－様々な帳票手法/CS/Common/Make広告.cs
using Pao.Reports;
using System;
using System.Data;
using System.Data.OleDb;

namespace Sample
{
    static class Make広告
    {
        public static void SetupData(IReport paoRep)
        {
            //Excel データベースへの接続
            OleDbConnection connection = Util.ConnectExcelDB("広告.xls");
            string sql = "select * from [広告情報$]";

            OleDbDataAdapter dataAdapter = new OleDbDataAdapter(sql, connection);
            DataSet ds = new DataSet();

            dataAdapter.Fill(ds, "広告情報");

            DataTable table = ds.Tables["広告情報"];

            paoRep.LoadDefFile(Util.SharePath + "広告.prepd");
            foreach (DataRow row in table.Rows)
            {

                paoRep.PageStart();

                paoRep.Write("製品名", (string)row["製品名"]);
                paoRep.Write("キャッチフレーズ", (string)row["キャッチフレーズ"]);
                paoRep.Write("商品コード", (string)row["商品コード"]);
                paoRep.Write("JANコード", (string)row["商品コード"]);
                paoRep.Write("売り文句", (string)row["売り文句"]);
                paoRep.Write("説明", (string)row["説明"]);
                paoRep.Write("価格", (string)row["価格"]);
                paoRep.Write("画像1", Util.SharePath + (string)row["画像1"]);
                paoRep.Write("画像2", Util.SharePath + (string)row["画像2"]);
                paoRep.Write("QR", (string)row["製品名"] + " " + (string)row["キャッチフレーズ"]);

                paoRep.PageEnd();

            }

            return;
        }
    }
}

File: 02.実践帳票サンプル集－様々な帳票手法/VB/Common/Make広告.vb
Imports Pao.Reports
Imports System
Imports System.Data
Imports System.Data.OleDb
Imports System.Windows

Namespace Sample
    Friend Module Make広告
        Public Sub SetupData(paoRep As IReport)
            'Excel データベースへの接続
            Dim connection As OleDbConnection = Util.ConnectExcelDB("広告.xls")
            Dim sql As String = "select * from [広告情報$]"
            Dim dataAdapter As New OleDbDataAdapter(sql, connection)
            Dim ds As New DataSet()

            dataAdapter.Fill(ds, "広告情報")

            Dim table As DataTable = ds.Tables("広告情報")
            paoRep.LoadDefFile(Util.SharePath & "広告.prepd")

            For Each row As DataRow In table.Rows
                paoRep.PageStart()
                paoRep.Write("製品名", CStr(row("製品名")))
                paoRep.Write("キャッチフレーズ", CStr(row("キャッチフレーズ")))
                paoRep.Write("商品コード", CStr(row("商品コード")))
                paoRep.Write("JANコード", CStr(row("商品コード")))
                paoRep.Write("売り文句", CStr(row("売り文句")))
                paoRep.Write("説明", CStr(row("説明")))
                paoRep.Write("価格", CStr(row("価格")))
                paoRep.Write("画像1", Util.SharePath & CStr(row("画像1")))
                paoRep.Write("画像2", Util.SharePath & CStr(row("画像2")))
                paoRep.Write("QR", CStr(row("製品名")) & " " & CStr(row("キャッチフレーズ")))
                paoRep.PageEnd()
            Next

            Return
        End Sub
    End Module
End Namespace